modules/roles/server/controllers/roles.controller.js

```diff
@@ -655,22 +677,26 @@ function roleModule(logger, userModule, moduleLoader) {
       }
     }
   }
-    
+ 
+
 
   // --------------------------- Revealing Module Section ----------------------------
 
-  return {
+  let publicFunctions = {
     create                    : addRole,
     list                      : listAllRoles,
     update                    : updateRole,
     delete                    : removeRole,
     subroles                  : getSubroles,
+    determineSubroles         : determineSubroles,
     tree                      : getRoleTree,
     updateUserRole            : updateUserRole,
     reportEndpointPermissions : reportEndpointPermissions,
     pruneEndpointDetails      : pruneEndpointDetails,
     updateSingleRole          : updateSingleRole
   }
+  userModule.crud.setRoleModule(publicFunctions);
```

Comment:
eww, was this necessary?  I am guessing the self thing didn't resolve completely?

Another option that would maybe look a little cleaner is to do this in the module's index.js.  You could get a handle on the roleModule post-creation there and fire a function off maybe?